### PR TITLE
Assert we have triton before scheduling on triton

### PR DESCRIPTION
**Pull request by voznesenskym**
cc @mlazos @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by wconstab — APPROVED** [review]
i think we need to rename the helper and update the messages, since some folks will hit this due to an old GPU and be confused why installing triton isn't helping.

**Review by ngimel — APPROVED** [review]

**Comment by voznesenskym** [comment]
> i think we need to rename the helper and update the messages, since some folks will hit this due to an old GPU and be confused why installing triton isn't helping.

Open to suggestions ;)

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -1078,6 +1078,7 @@ def create_backend(self, device: torch.device):
 
             return CppScheduling(self)
         else:
+            assert has_triton(), f"Found {device.type}, but triton is not installed. Please install triton."
```
Comment (soumith) [inline torch/_inductor/scheduler.py:1081]:
```
device_props = torch.cuda.get_device_properties(device)

if device_props.major < 6:
    RuntimeError(f"Found {device_props.name} which is too old to be supported by the triton GPU compiler, which is used as the backend. Triton only supports devices of CUDA Capability >= 6.0, but your device is of CUDA capability {device_props.major}.{device_props.minor}")
elif not has_triton():
    RuntimeError(f"Cannot find a working triton installation. More information on installing Triton can be found at https://github.com/openai/triton")
```

Comment (voznesenskym) [reply torch/_inductor/scheduler.py:1081]: Thanks @soumith, updated

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**